features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/TimelineView.kt

```diff
@@ -161,28 +161,20 @@ fun TimelineItemRow(
             )
         }
         is TimelineItem.Event -> {
-            fun onClick() {
-                onClick(timelineItem)
-            }
-
-            fun onLongClick() {
-                onLongClick(timelineItem)
-            }
-
             if (timelineItem.content is TimelineItemStateContent) {
                 TimelineItemStateEventRow(
                     event = timelineItem,
                     isHighlighted = highlightedItem == timelineItem.identifier(),
-                    onClick = ::onClick,
-                    onLongClick = ::onLongClick,
+                    onClick = { onClick(timelineItem) },
+                    onLongClick = { onLongClick(timelineItem) },
```

Comment:
So the TimelineItem passed in the callback isn't right? Should we change the callback to just be a `() -> Unit` in that case? :thinking:

Reply:
I'm not 100% sure on my take [here](https://github.com/vector-im/element-x-android/pull/774#discussion_r1252360667), but I think that's probably what's happening with these function references.

Reply:
This already happened in `ActionListView`, and switching the function reference back to a lambda fixed the issue, returning the current value instead.